pkg/util/admission/work_queue.go

```diff
@@ -456,8 +465,30 @@ type tenantHeap []*tenantInfo
 
 var _ heap.Interface = (*tenantHeap)(nil)
 
+var tenantInfoPool = sync.Pool{
+	New: func() interface{} {
+		return &tenantInfo{}
+	},
+}
+
 func makeTenantInfo(id uint64) *tenantInfo {
```

Comment:
nit: I'd call this `getTenantInfo` or `newTenantInfo`. The convention seems to be `make*` for non-pointer structs and `new*` for pointers. I've used `get` as the constructor prefix when getting from a pool. Same nit would apply to `makeWaitingWork`. Take it or leave it.